### PR TITLE
Add Authorization rules

### PR DESCRIPTION
**Pull request by danielmatama**
This PR, we implemented following features:

Installed CanCanCan to project.

 Add a role column to the users table.

 A user can delete a post if it is theirs or if they have an admin role (column role has value "admin").

 Add the "Delete" button to the view and make sure that only authorized users can see it.

 A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin").

 Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :manage, :all if user.role == 'admin'
+
+    can :destroy, Post, author: user
+    can :destroy, Comment, author: user
```
Comment (Ismailco) [inline app/models/ability.rb:8]: - [ ] [OPTIONAL] The owner of a post can perform all view, update, or destroy their own post if Post.author == user, kindly consider changing `destroy` to `manage` for this to work as expected

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+class PostsController < ApplicationController
```
Comment (Ismailco) [inline app/controllers/posts_controller.rb:1]: - [ ] Kindly consider using cancan authorization in the `Post controller` by adding `load_and_authorize_resource` to the controller

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @danielmatama, @AmanWorku, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AmanWorku** [comment]
Thank you @Ismailco.

**Review by muneebulrehman — APPROVED** [review]
Hi @danielmatama , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- Good follow-up with the previous reviewer

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AmanWorku** [comment]
@muneebulrehman Thank you!

**Pull request merged**